lib/public_key/src/pubkey_os_cacerts.erl

```diff
@@ -0,0 +1,184 @@
+            persistent_term:get(?MODULE);
+        CaCerts ->
+            CaCerts
+    end.
+
+%% (Re)Load default os cacerts and cache result.
+-spec load() ->  ok | {error, Reason::term()}.
+load() ->
+    case os:type() of
+        {unix, linux} ->
+            load_from_file(linux_paths());
+        {unix, openbsd} ->
+            load_from_file(bsd_paths());
+        {unix, freebsd} ->
+            load_from_file(bsd_paths());
+        {unix, sunos} ->
+            load_from_files(sunos_path());
+        {win32, _} ->
+            load_win32();
+	{unix, darwin} ->
+            load_darwin();
+        Os ->
+            {error, {enotsup, Os}}
+    end.
+
+%% (Re)Load cacerts from file and cache result.
+%% The file-paths will be tried in order.
+%% Can be used when load/0 doesn't work for an unsupported os type.
+-spec load([file:filename_all()]) -> ok | {error, Reason::term()}.
+load(Paths) ->
+    load_from_file(Paths).
+
+%% Implementation
+load_from_file([Path|Paths]) when is_list(Path); is_binary(Path) ->
+    try
+        {ok, Binary} = file:read_file(Path),
+        ok = decode_result(Binary)
+    catch _:_Reason ->
+            load_from_file(Paths)
+    end;
+load_from_file([]) ->
+    {error, enoent}.
+
+decode_result(Binary) ->
+    try
+        MakeCert = fun({'Certificate', Der, not_encrypted}, Acc) ->
+                           try
+                               Decoded = public_key:pkix_decode_cert(Der, otp),
+                               [#cert{der=Der, otp=Decoded}|Acc]
+                           catch _:_ ->
+                                   Acc
+                           end
+                   end,
+        Certs = lists:foldl(MakeCert, [], pubkey_pem:decode(Binary)),
+        store(Certs)
+    catch _:Reason ->
+            {error, Reason}
+    end.
+
+
+load_from_files(Path) ->
+    MakeCert = fun(FileName, Acc) ->
+                       try
+                           {ok, Bin} = file:read_file(FileName),
+                           [#cert{der=Der, otp=public_key:pkix_decode_cert(Der, otp)}
+                            || {'Certificate', Der, not_encrypted} <- pubkey_pem:decode(Bin)]
+                               ++ Acc
+                       catch _:_ ->
+                               Acc
+                       end
+               end,
+    Certs = filelib:fold_files(Path, ".*\.pem", false, MakeCert, []),
+    store(Certs).
+
+
+load_win32() ->
+    Dec = fun({_Enc, Der}, Acc) ->
+                  try
+                      Decoded = public_key:pkix_decode_cert(Der, otp),
+                      [#cert{der=Der, otp=Decoded}|Acc]
+                  catch _:_ ->
+                          Acc
+                  end
+          end,
+    store(lists:foldl(Dec, [], os_cacerts())).
+
+load_darwin() ->
+    %% Temporary solution, should probably be re-written to use Keychain Access API
+    KeyChainFile = "/System/Library/Keychains/SystemRootCertificates.keychain",
+    TmpPemFile = "/tmp/all_certs.pem",
```

Comment:
Probably not, this should be good enough, tm.
Please do, this will remain until someone with Obj-C knowledge decides that it is a much better solution,
and updates the nif, writes a configure file and ..

Reply:
Does the `security export` command's `-o` switch allow `-` as an argument to export to stdout, removing the use of a temp file? I'd check but I don't have access to OS X.

Reply:
Going through Security API would be ideal but in the meantime I think we can improve how we shell out. The default is to write to stdout so going through the temp file should be unnecessary:

```erl
> X = os:cmd("security export -t certs -f pemseq -k /System/Library/Keychains/SystemRootCertificates.keychain"), {lists:sublist(X,10), length(X)}.
{"-----BEGIN",249291}
```